### PR TITLE
[Wasm] Reduce max length on browser for ToArrayShouldWorkWithSpecialLengthLazyEnumerables Linq test

### LINKED ISSUE #41114: Linq tests timing out in WASM leg
Linq tests appear to be timing out (or possibly hanging) in WASM

https://dev.azure.com/dnceng/public/_build/results?buildId=780635&view=ms.vss-test-web.build-test-results-tab&runId=24492990&resultId=164319&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab
https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-41038-merge-558fe2197ef6431186/System.Linq.Tests/console.01a47de4.log?sv=2019-02-02&se=2020-09-09T20%3A59%3A51Z&sr=c&sp=rl&sig=Nv2ILnn693jzHBwsZrPiM8NJrLkYK7Qe2su3kYUaY78%3D

```
Executed on a003RYO
+ export PATH=/home/helixbot/work/A66808D0/p/dotnet:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
+ export DOTNET_ROOT=/home/helixbot/work/A66808D0/p/dotnet
+ export DOTNET_CLI_TELEMETRY_OPTOUT=1
+ export PATH=/home/helixbot/work/A66808D0/p/xharness-cli:/home/helixbot/work/A66808D0/p/dotnet:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
+ export XHARNESS_DISABLE_COLORED_OUTPUT=true
+ export XHARNESS_LOG_WITH_TIMESTAMPS=true
+ export XHARNESS_CLI_PATH=/home/helixbot/work/A66808D0/p/microsoft.dotnet.xharness.cli/1.0.0-prerelease.20411.1/tools/netcoreapp3.1/any/Microsoft.DotNet.XHarness.CLI.dll
+ ./RunTests.sh --runtime-path /home/helixbot/work/A66808D0/p
XHarness command issued: wasm test --engine=V8 --engine-arg=--stack-trace-limit=1000 --js-file=runtime.js -v --output-directory=./xharness-output -- --run WasmTestRunner.dll System.Linq.Tests.dll -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
[21:00:14] info: test[0]
      21:00:14.6178730 v8 --expose_wasm --stack-trace-limit=1000 runtime.js -- --run WasmTestRunner.dll System.Linq.Tests.dll -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing
      
[21:00:14] dbug: test[0]
      Arguments: --run,WasmTestRunner.dll,System.Linq.Tests.dll,-notrait,category=IgnoreForCI,-notrait,category=OuterLoop,-notrait,category=failing
      
[21:00:15] dbug: test[0]
      console.debug: MONO_WASM: Initializing mono runtime
      
[21:00:15] dbug: test[0]
      console.debug: MONO_WASM: ICU data archive(s) loaded, disabling invariant mode
      
[21:00:15] dbug: test[0]
      console.debug: mono_wasm_runtime_ready fe00e07a-5519-4dfe-b35a-f867dbaf2e28
      
[21:00:15] dbug: test[0]
      Initializing.....
      
[21:00:16] dbug: test[0]
      Discovering tests for System.Linq.Tests.dll
,...................

[21:15:10] dbug: test[0]
      [PASS] System.Linq.Tests.SkipLastTests.SkipLast(source: [0, 1, 2, 3, 4, ...], count: 8)
      
[21:15:12] dbug: test[0]
      [PASS] System.Linq.Tests.SkipLastTests.SkipLast(source: [0, 1, 2, 3, 4, ...], count: 13)
      
[21:15:13] dbug: test[0]
      [PASS] System.Linq.Tests.SkipLastTests.SkipLast(source: [0, 1, 2, 3, 4, ...], count: 21)
      
Killed
```

### PR DESCRIPTION
**Pull request by steveisok**
Addresses the occasional timeouts in System.Linq tests.  

The last 3-4 iterations of ToArrayShouldWorkWithSpecialLengthLazyEnumerables take between 20 - 56 seconds each and likely trigger longer GC runs.

Fixes https://github.com/dotnet/runtime/issues/41114

### CODE REVIEW

File: src/libraries/System.Linq/tests/ToArrayTests.cs [hunk 1]
```diff
@@ -331,6 +331,9 @@ public void NonConstantTimeCountEmptyPartitionSelectDiffTypeToArray()
         }
 
         [Theory]
+#if OUTERLOOP
```
Comment (jkotas) [inline src/libraries/System.Linq/tests/ToArrayTests.cs:334]: It should be fine move it to the outer loop everywhere. No need to special case it.

File: src/libraries/System.Linq/tests/System.Linq.Tests.csproj [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent)</TargetFrameworks>
+    <DefineConstants Condition="'$(TargetOS)' == 'Browser'">$(DefineConstants);OUTERLOOP</DefineConstants>
```
Comment (jkotas) [inline src/libraries/System.Linq/tests/System.Linq.Tests.csproj:4]: This would need Browser in the list of the TargetFrameworks above to work well.

Comment (danmosemsft) [reply src/libraries/System.Linq/tests/System.Linq.Tests.csproj:4]: I suggest to unilaterally outer loop instead

**Comment by akoeplinger** [comment]
I took a look at the Kusto data and even for slow runs of the System.Linq.Tests the really slow ToArrayShouldWorkWithSpecialLengthLazyEnumerables were ones with very large array lengths.

E.g. from this System.Linq.Tests run that took ~12mins overall:

| Arguments | Duration |
|-----------|----------|
| length: 106495	|1.1870459|
| length: 106496	|1.1880259|
| length: 106497	|1.2339749|
| length: 229375	|2.6252509|
| length: 229376	|2.5778229|
| length: 229377	|2.5524089|
| length: 491519	|5.391286|
| length: 491520	|5.408627|
| length: 491521	|5.566281|
| length: 1048575|	11.7551888|
| length: 1048576|	11.3395649|
| length: 1048577|	11.4221359|
| length: 2228223|	24.018349|
| length: 2228224|	23.8259399|
| length: 2228225|	23.6335699|
| length: 4718591|	50.1064809|
| length: 4718592|	50.2593909|
| length: 4718593|	50.277007|

I think the better way is to just exclude lengths of  > 1000000 from the test run on Browser.

**Review by akoeplinger — APPROVED** [review]

**Review by safern — COMMENTED** [review]
LGTM

**Pull request merged**